### PR TITLE
fix: Add LegalBench datasets - 15

### PR DESCRIPTION
**Pull request by awinml**
## Checklist for adding MMTEB dataset

Reason for dataset addition: Showcases the model's performance on the [LegalBench](https://hazyresearch.stanford.edu/legalbench/) datasets, enabling evaluation on domain-specific legal texts.

Follow up to #680 

## Changes Made

The following 10 datasets have been added in this PR:

**Classification:**

1. [OPP115 Policy Change](https://hazyresearch.stanford.edu/legalbench/tasks/opp115_policy_change.html)
2. [OPP115 Third Party Sharing Collection](https://hazyresearch.stanford.edu/legalbench/tasks/opp115_third_party_sharing_collection.html)
3. [OPP115 User Access Edit And Deletion](https://hazyresearch.stanford.edu/legalbench/tasks/opp115_user_access,_edit_and_deletion.html)
4. [OPP115 User Choice Control](https://hazyresearch.stanford.edu/legalbench/tasks/opp115_user_choice_control.html)
5. [Oral Argument Question Purpose](https://hazyresearch.stanford.edu/legalbench/tasks/oral_argument_question_purpose.html)
6. [Overruling](https://hazyresearch.stanford.edu/legalbench/tasks/overruling.html)
7. [Personal Jurisdiction](https://hazyresearch.stanford.edu/legalbench/tasks/personal_jurisdiction.html)
8. [PROA](https://hazyresearch.stanford.edu/legalbench/tasks/proa.html)

**Pair Classification:**

   9. [Privacy Policy Entailment](https://hazyresearch.stanford.edu/legalbench/tasks/privacy_policy_entailment.html)
  10. [Privacy Policy QA](https://hazyresearch.stanford.edu/legalbench/tasks/privacy_policy_qa.html)

-------------------------------------------------------------------------------------------------------------------

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: results/sentence-transformers__paraphrase-multilingual-MiniLM-L12-v2/PrivacyPolicyEntailmentLegalBenchPC.json
Comment (KennethEnevoldsen) [inline results/sentence-transformers__paraphrase-multilingual-MiniLM-L12-v2/PrivacyPolicyEntailmentLegalBenchPC.json:1]: Very similar scores here. Will you run e5-base? (or some legal embedding model)

Comment (awinml) [reply results/sentence-transformers__paraphrase-multilingual-MiniLM-L12-v2/PrivacyPolicyEntailmentLegalBenchPC.json:1]: As discussed in https://github.com/embeddings-benchmark/mteb/pull/693#issuecomment-2109696771, I have combined the `PrivacyPolicy` Datasets into `LegalBenchPC`. The scores are not identical anymore, they are close, but still significantly different. I have also included the scores using the e5-base model for comparison.

File: docs/mmteb/points/682.jsonl [hunk 1]
```diff
@@ -0,0 +1 @@
+{"GitHub": "awinml", "New dataset": 20}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/682.jsonl:1]:
```suggestion
{"GitHub": "kennethenevoldsen", "PR Review": 2}
```

**Review by KennethEnevoldsen — APPROVED** [review]

File: results/sentence-transformers__paraphrase-multilingual-MiniLM-L12-v2/PrivacyPolicyEntailmentLegalBenchPC.json
Comment (KennethEnevoldsen) [reply results/sentence-transformers__paraphrase-multilingual-MiniLM-L12-v2/PrivacyPolicyEntailmentLegalBenchPC.json:1]: Thanks!

File: docs/mmteb/points/682.jsonl [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "kennethenevoldsen", "PR Review": 2}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/682.jsonl:1]:
```suggestion
{"GitHub": "awinml", "New dataset": 20}
```

**Review by KennethEnevoldsen — APPROVED** [review]

**Pull request merged**